features/step_definitions/logging.rb

```diff
@@ -1056,3 +1056,125 @@
     | pod_label       | #{pod_label}     |
   })
 end
+
+
+
+Given /^I have(?: "(\w+)")? log pod in project #{QUOTED}$/ do | log_type, project_name |
+    log_type ||= "any"
+    file_dir = "#{BushSlicer::HOME}/testdata/logging/loggen"
+
+    unless project(project_name).exists?
+        step %Q/I run the :new_project client command with:/,table(%{
+          | project_name | #{project_name} |
+        })
+        raise "Error creating namespace" unless @result[:success]
+    end
+
+    case log_type
+      when "json"
+        template_file = "#{file_dir}/container_json_log_template.json"
+      when "flat"
+        template_file = "#{file_dir}/container_json_unicode_log_template.json"
+      when "unicode"
+        template_file = "#{file_dir}/container_json_unicode_log_template.json"
+      else
+        template_file = "#{file_dir}/container_json_log_template.json"
+      end
+
+    unless  project.pods(by:user).size > 0 
+        step %Q/I run the :new_app client command with:/,table(%{
+           | file | #{template_file} |
+         })
+    end
+end
+
+Given /^I have index pattern #{QUOTED}$/ do | pattern_name |
+    step %Q/I perform the :kibana_index_pattern_exist web action with:/,table(%{
+        | index_pattern_name | #{pattern_name} |
+    })
+    unless @result[:success]
+        step %Q/I run the :go_to_kibana_management_page web action/
+        step %Q/I perform the :create_index_pattern_in_kibana web action with:/, table(%{
+            | index_pattern_name | "#{pattern_name}" |
+         })
+         raise "Failed to create index pattern #{pattern_name}" unless @result[:success]
+    end
+end
+
+Given /^cluster-admin create the #{QUOTED} pattern for the #{WORD} user$/ do | pattern_name, who |
```

Comment:
@anpingli Could you please update the step name? `cluster-admin create xxxx` can easily make some misunderstanding here.

Reply:
@QiaolingTang   How about openshift-Logging admin who can modify the ES directly?

Reply:
How about `create index pattern via rest api`?

Reply:
The step must be running as project admin.  we use cluster-admin here.  I think it is better to keep some works like 'cluster admin, logging admin'.   so we can know from the name that we must switch to cluster-admin before call this step.

Reply:
Any keywords to make that clear if the step need the admin role or not?

Reply:
> The step must be running as project admin. I think it is better to keep some works like 'cluster admin, logging admin'. so we can know from the name that we must switch to cluster-admin before call this step.

I see, thank you.
BTW, it switches to cluster-admin in this step, so no need to switch to cluster-admin before calling this step.
Besides, it seems this step is not used.